### PR TITLE
fix: make brand optional in the types of `getAmountAllocated`

### LINKED ISSUE #1725: zcf.getAmountAllocated should either assert a brand argument or allow it to be optional
**What is the Problem Being Solved?**

We use the second parameter to `getAmountAllocated` (brand) to fill in an empty amount if the keyword is not found. This means it is only used if the keyword is not found. However, we require it in the types. This seems inconsistent.  For example, in the below test, the first test is fine despite the brand not being passed, but the third test is not ok, because the keyword didn't exist and we actually tried to use the brand. 

Screen Shot 2020-09-08 at 6 49 17 PM

**Description of the Design**

We should be consistent: either make the brand optional (my preference) or assert its existence at the top of the function.

**Security Considerations**

Less confusion about usage.

**Test Plan**
ZCF tests

### PR DESCRIPTION
**Pull request by katelynsills**
Closes #1725

### CODE REVIEW

File: packages/zoe/src/contractFacet/seat.js [hunk 1]
```diff
@@ -73,6 +73,7 @@ export const makeZcfSeatAdminKit = (
       if (currentAllocation[keyword] !== undefined) {
         return currentAllocation[keyword];
       }
+      assert(brand, `A brand must be supplied if the keyword may not exist`);
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/seat.js]:
```suggestion
      assert(brand, `A brand must be supplied when the keyword is not defined`);
```
"may not" seems wrong. It's only required (by the implementation) when the keyword is not found. As advice to callers, I might say "when you're not sure that the keyword will be present, you should include the brand", but that's about their state of mind, not what is enforced.

Also, "exist" seems like the wrong verb. How do you feel about "is not present", "is not found" or "is not defined"?

**Review by Chris-Hibbert — COMMENTED** [review]
Good change. I have questions about the error message.

**Review by Chris-Hibbert — APPROVED** [review]
sorry, that was supposed to be approval. I have questions, but leave the resolution to your judgement.

File: packages/zoe/src/contractFacet/seat.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/contractFacet/seat.js]: This new error message looks good to me. thanks!

**Pull request merged**